devdoc.md

```diff
@@ -0,0 +1,165 @@
+
+#### Installing package dependencies using R functions
+
+Installing the sourmashconsumr package from github will trigger missing dependencies to be installed.
+
+```
+install.packages("devtools")
+devtools::install_github("Arcadia-Science/sourmashconsumr")
+```
+
+Some of the package dependencies are available on Bioconductor (namely phyloseq), so if installation of dependencies fails through `install_github`, you may have more success using `BiocManager`:
+```
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+
+BiocManager::install("Arcadia-Science/sourmashconsumr")
+```
+
+#### Managing the development environment with conda
+
+If the above methods fail, conda can be used to install the dependencies.
+You can find operating system-specific instructions for installing miniconda [here](https://docs.conda.io/en/latest/miniconda.html).
+Once you have conda and [mamba](https://mamba.readthedocs.io/en/latest/) installed, you can create an environment with all of the sourmashconsumr dependencies and developer packages installed using the following bash command:
+
+```
+mamba env create -n sourmashconsumr --file devenv.yml
+```
+
+Then, activate the environment:
+
+```
+conda activate sourmashconsumr
+```
+
+You will still need to install RStudio after creating the conda environment. 
+Then, if you're using a mac computer, you can open the sourmashconsumr Rproject by running `open sourmashconsumr.Rproj` from the activated environment you just created.
+
+```
+open sourmashconsumr.Rproj
+```
+
+On a linux or windows, running `rstudio` from the activated conda environment may open RStudio (untested).
+
+```
+rstudio
+```
+
+## Developing 
+
+sourmashconsumr is developed on GitHub at [Arcadia-Science/sourmashconsumr](https://github.com/Arcadia-Science/sourmashconsumr/).
+Changes can be contributed to sourmashconsumr via [Pull Request](https://github.com/Arcadia-Science/sourmashconsumr/pulls).
+
+### Getting started
+
+To start developing, open the sourmashconsumr.Rproj in RStudio (see above).
+Then, load the `devtools` package and use the `devtools::load_all()` function to load sourmashconsumr.
+
+```
+library(devtools)
+load_all()
+```
+
+You can then open the file you want to change and make changes.
+Core package functions can be found in the `R` folder. 
+Functions are grouped by action (e.g. `read_sourmash_outputs.R` contains all of the `read_*()` functions) or by data type (e.g. functions that involve plotting the output of sourmash taxonomy annotate are in the `plot_taxonomy_annotate.R` file).
+Tests and test files are located in the `tests/testthat` folder.
+
+### Tests
+
+The sourmashconsumr package uses unit tests to make sure that code changes don't break existing functions.
+To run all tests, you can use the `testthat::test()` function:
+
+```
+test()
+```
+
+Alternatively, you can run tests related to the active file:
+```
+test_active_file()
+```
+
+### Documentation
```

Comment:
I have only done this once and it was a tad hacky, but I think quarto or a different R package can make it easy to create a static site from the existing documentation so that it's easier to search through and browse (packagedown I think?)

Reply:
ya packagedown makes websites! I thought about going for that, but I decided to hold off because I wasn't sure if I wanted a whole separate documentation website yet. I might take the plunge later this week and then I'd transfer all of this stuff over. good to hear your opinion though, thank you!